### PR TITLE
Update Sphinx cheatsheet in AiiDA docs

### LINKED ISSUE #3180: Check documentation for "for AiiDA developers/Sphinx cheatsheet"
Check if the section is updated to AiiDA 1.0.

https://aiida.readthedocs.io/projects/aiida-core/en/latest/developer_guide/tools/sphinx_cheatsheet.html

### PR DESCRIPTION
**Pull request by CasperWA**
Fixes #3180 

Related to #3034 

Removing `index.rst` from developers_guide, and copied over the toc of
the `index.rst` that defines the menu on the left of the documentation.

This is done to remove the `<no-title>` from the "breadcrumbs" at the
top of all the pages under the section "For AiiDA developers".

### CODE REVIEW

File: docs/source/developer_guide/index.rst [hunk 1]
```diff
@@ -1,17 +0,0 @@
-Practical information concerning how to set up your development environment and how
-to contribute to aiida core can be found on the 
-`AiiDA Wiki <https://github.com/aiidateam/aiida-core/wiki>`_.
```
Comment (ltalirz) [inline docs/source/developer_guide/index.rst]: is this information still present at a prominent position?

Comment (CasperWA) [reply docs/source/developer_guide/index.rst]: No, unfortunately not.
However, I think perhaps it should be moved to the front page, under a special section "For developers"?

Comment (CasperWA) [reply docs/source/developer_guide/index.rst]: Also note that this particular page was basically unreachable before - unless somebody pressed the `<no title>` "breadcrumb".

Comment (ltalirz) [reply docs/source/developer_guide/index.rst]: could you please preserve this information in a place where it makes sense?

**Comment by CasperWA** [comment]
I have added the reference to the AiiDA Wiki on the landing page of the documentation under a new section "For developers".

**Pull request merged**